### DISCUSSION TITLE
How to generate sitemaps.xml using react-router-dom?

### DISCUSSION THREAD

**Discussion by Haiz14** [remix-run/react-router#10419] [Q&A]
Creating an spa and just wanted to generate the sitemap.xml.

**Comment by MeiKatz** [comment]
You can only get a list of all routes if you define all your routes in one place. But RR cannot replace all possible replacements of placeholders for you. Maybe the best way is to generate them on the server-side by yourself.

**Reply by ghost** [reply]
I defined all my routes in one router.jsx, now how do i generate sitemap.xml
